### PR TITLE
E2E: plant search tests

### PR DESCRIPTION
**Pull request by 4ydan**
A small set of plant search tests. Lmk if any scenario is missing. I think this covers the basic idea.

## Basics

- [x] I added a line to [/doc/CHANGELOG.md](/doc/CHANGELOG.md)
- [x] The PR is rebased with current master.
- [x] Details of what you changed are in commit messages.
- [x] References to issues, e.g. `close #X`, are in the commit messages.
- [x] The buildserver is happy.

## Checklist

- [ ] I have installed and I am using [pre-commit hooks](../doc/contrib/README.md#Hooks)
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I fixed the introduction tour
- [ ] I wrote migrations in a way that they are compatible with already present data
- [ ] I fixed all affected decisions
- [ ] I added unit tests for my code
- [ ] I added code comments, logging, and assertions as appropriate
- [ ] I translated all strings visible to the user
- [ ] I mentioned [every code or binary](https://github.com/ElektraInitiative/PermaplanT/blob/master/.reuse/dep5) not directly written or done by me in [reuse syntax](https://reuse.software/)
- [ ] I created left-over issues for things that are still to be done
- [ ] Code is conforming to [our Architecture](/doc/architecture)
- [ ] Code is conforming to [our Guidelines](/doc/guidelines)
      (exceptions are documented)
- [ ] Code is consistent to [our Design Decisions](/doc/decisions)

## Review

- [ ] I've tested the code
- [ ] I've read through the whole code
- [ ] Examples are well chosen and understandable

### CODE REVIEW

File: e2e/features/search_plants.feature [hunk 1]
```diff
@@ -0,0 +1,18 @@
+Feature: Searching for Plants
+
+  Background:
+    Given I am on the SUTSearching map page and I have selected the plant layer
+
+  Scenario Outline: Searching for plants with partial matches
+    When I type <plant> into the search box
+    Then the app should display <result>
```
Comment (markus2330) [inline e2e/features/search_plants.feature:8]: unclear if it is first result or any result

```suggestion
    Then the app should display <result> as first match
```

File: e2e/pages/maps/planting.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+class MapPlantingPage(AbstractPage):
+    """The planting page of a map on permaplant"""
+    TITLE: str = 'PermaplanT'
+
+    def __init__(self, page: Page):
+        self.page = page
+        self.base_layer_radio = self.page.get_by_role("radio", name="select layer").first
+        self.plant_layer_radio = self.page.get_by_role("radio", name="select layer").nth(1)
+        self.plant_search_icon = self.page.locator(".flex-shrink > div > .flex > .inline-flex").first
```
Comment (markus2330) [inline e2e/pages/maps/planting.py:13]: this doesn't look very stable, can you add some data-testid there?

File: e2e/pages/maps/planting.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        self.page.get_by_role("button", name=plantname).click()
+
+    def close_tour(self):
+        """
+        Currently you get greeted with the tour.
+        In case you rerun the tests locally, this should not fail.
+        """
+        try:
+            self.tour_close = self.page.get_by_label("Close Tour")
+            self.tour_close.click()
```
Comment (markus2330) [inline e2e/pages/maps/planting.py:40]: #753 will introduce a confirmation

File: e2e/steps/test_map_creation.py [hunk 4]
```diff
@@ -21,6 +21,9 @@ def logged_in_and_on_map_management_page(hp: HomePage, lp: LoginPage):
     hp.to_map_management_page()
 
 
+# Scenario 1
```
Comment (markus2330) [inline e2e/steps/test_map_creation.py:24]: Better give names?

File: e2e/steps/test_plant_search.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+from e2e.pages.maps.planting import MapPlantingPage
+from e2e.pages.maps.management import MapManagementPage
+from pytest_bdd import scenarios, given, when, then, parsers
+
+scenarios("features/search_plants.feature")
+
+
+@given(parsers.parse("I am on the {name} map page and I have selected the plant layer"))
+def logged_in(hp: HomePage, lp: LoginPage, mmp: MapManagementPage, mcp: MapCreatePage, mpp: MapPlantingPage, name):
+    # Login
```
Comment (markus2330) [inline e2e/steps/test_plant_search.py:14]: I think we should have a function login and reuse it.

File: e2e/steps/test_plant_search.py [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+
+@when(parsers.parse("I type {plant} into the search box"))
+def search_a_plant(mpp: MapPlantingPage, plant):
+    mpp.click_search_icon()
+    mpp.fill_plant_search(plant)
+
+
+@then(parsers.parse("the app should display {result}"))
+def create_map_successfully(mpp: MapPlantingPage, result):
```
Comment (markus2330) [inline e2e/steps/test_plant_search.py:43]: name is misleading, probably copy&paste error?

File: e2e/pages/maps/planting.py [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+        """Clicks on the canvas. Default to x300 y300"""
+        self.page.locator("canvas:nth-child(6)").click(position={"x": x,"y": y})
+
+    def click_delete(self):
+        """Deletes a planting by pressing on the delete button"""
+        self.delete_plant_button.click()
+
+    def search_result_is_visible(self, result):
+        """Confirm a search result is visible"""
+        expect(self.page.get_by_role("button", name=result)).to_be_visible()
```
Comment (markus2330) [inline e2e/pages/maps/planting.py:58]: "button" also doesn't sound very robust.

File: e2e/pages/maps/planting.py [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+
+    def no_plants_matching_query(self):
+        """Confirm no plants are found is shown"""
+        self.page.get_by_text("There are no plants matching your search query...")
```
Comment (markus2330) [inline e2e/pages/maps/planting.py:62]: Also not very stable, at least use regex or imho even better data-testid.

**Review by markus2330 — CHANGES_REQUESTED** [review]
Very nicely written test. Some locators etc. seem a bit shaky to me.

File: e2e/pages/maps/planting.py (hunk 2 above)
Comment (4ydan) [reply e2e/pages/maps/planting.py:13]: I see there is already many `data-tourid`s on elements I'm also testing. Do I just duplicate my testid's now or should I reuse those, any specific ideas about that?

**Comment by 4ydan** [comment]
Btw, I m not sure if its very user friendly when the search gets deleted/reset after picking a plant. Was that planned?

**Comment by markus2330** [comment]
Thx for telling! You are right, it is not wanted like this, @yvonnemarkl actually already told me she would like to have it otherwise. The main usability problem is that  you can lose the plant you are planting by accident, and then you need to search for the same plant again. Can you please open an issue? If we change that searches do not get deleted, we probably need a "text delete" button for convenience.

File: e2e/pages/maps/planting.py (hunk 2 above)
Comment (markus2330) [reply e2e/pages/maps/planting.py]: IMHO same as other ids, there might be some internal use. So I would continue with our general guideline: either user-facing text or `data-testid`. Please write this clearly in #653. But let Bushuo also review  #653 after you finalized it from your point of view.

**Comment by 4ydan** [comment]
Ready to be merged 🚀

File: e2e/ReadMe.md
Comment (markus2330) [inline e2e/ReadMe.md:1]: Should be called README.md like the others.

**Review by markus2330 — APPROVED** [review]
I like it! :heart:

**Pull request merged**